src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go

```diff
@@ -78,82 +88,41 @@ func (a *StorageMinerActorCode_I) NotifyOfSurprisePoStChallenge(rt Runtime) Invo
 	a._expirePreCommittedSectors(rt)
 
 	h, st := a.State(rt)
-	st.ChallengeStatus().Impl().LastChallengeEpoch_ = rt.CurrEpoch()
+	// update challenge start epoch
+	st.ChallengeStatus().Impl().OnNewChallenge(rt.CurrEpoch())
 	UpdateRelease(rt, h, st)
 	return rt.SuccessReturn()
 }
 
-func (a *StorageMinerActorCode_I) _slashDealsFromFaultReport(rt Runtime, sectorNumbers []sector.SectorNumber, action deal.StorageDealSlashAction) {
-
-	h, st := a.State(rt)
-
-	dealIDs := make([]deal.DealID, 0)
-
-	for _, sectorNo := range sectorNumbers {
-
-		utilizationInfo := st._getUtilizationInfo(rt, sectorNo)
-		activeDealIDs := utilizationInfo.DealExpirationAMT().Impl().ActiveDealIDs()
-		dealIDs = append(dealIDs, activeDealIDs...)
-
-	}
+// Called by the cron actor at every tick.
+func (a *StorageMinerActorCode_I) OnCronTick(rt Runtime) InvocOutput {
+	a.CheckSurprisePoStSubmissionHappened(rt)
 
-	Release(rt, h, st)
-
-	// TODO: Send(StorageMarketActor, ProcessDealSlash)
-}
-
-// construct FaultReport
-// reset NewTerminatedFaults
-func (a *StorageMinerActorCode_I) _submitFaultReport(
-	rt Runtime,
-	newDeclaredFaults sector.CompactSectorSet,
-	newDetectedFaults sector.CompactSectorSet,
-	newTerminatedFaults sector.CompactSectorSet,
-) {
-	faultReport := &sector.FaultReport_I{
-		NewDeclaredFaults_:   newDeclaredFaults,
-		NewDetectedFaults_:   newDetectedFaults,
-		NewTerminatedFaults_: newTerminatedFaults,
-	}
-
-	rt.Abort("TODO") // TODO: Send(SPA, ProcessFaultReport(faultReport))
-	panic(faultReport)
-
-	// only terminatedFault will be slashed
-	if len(newTerminatedFaults) > 0 {
-		a._slashDealsFromFaultReport(rt, newTerminatedFaults.SectorsOn(), deal.SlashTerminatedFaults)
-	}
-
-	h, st := a.State(rt)
-	st.SectorTable().Impl().TerminatedFaults_ = sector.CompactSectorSet(make([]byte, 0))
-	UpdateRelease(rt, h, st)
+	return rt.SuccessReturn()
 }
 
-// construct PowerReport from SectorTable
-// need lastPoSt to search for new expired deals in _updateSectorUtilization
-// where DealExpirationAMT takes in a range of Epoch and return a list of values that expire in that range
-func (a *StorageMinerActorCode_I) _submitPowerReport(rt Runtime, lastPoSt block.ChainEpoch) {
-	h, st := a.State(rt)
-	newExpiredDealIDs := st._updateSectorUtilization(rt, lastPoSt)
-	activePower := st._getActivePower(rt)
-	inactivePower := st._getInactivePower(rt)
+// If the miner is currently challenged for surprise PoSt,
+// reports every sector as failing for the current proving period.
+func (a *StorageMinerActorCode_I) CheckSurprisePoStSubmissionHappened(rt Runtime) InvocOutput {
```

Comment:
So the comment is right now, but it still seems very wrong to me that this is called on every cron tick. The miner will be instantly faulted with no window to compute PoSt.

Reply:
Now I've seen the largeer issue. Sorry I missed your point earlier. I believe the `ExpirePreCommitted` is as intendeed per @zixuanzh. The rest is indeed faulty. Fixed

Reply:
The comment could be tweaked for the new better semantics.

Reply:
@anorth i think it is okay that miner is immediately faulted when they have missed their SurprisePoSt submission window. It is a separate issue if we think calling this at every Cron tick is too expensive. @jbenet previously thinks this is okay.